deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/iterator/MultipleEpochsIterator.java

```diff
@@ -211,7 +211,7 @@ public void setPreProcessor(org.nd4j.linalg.dataset.api.DataSetPreProcessor preP
      */
     @Override
     public boolean hasNext() {
-        if (newEpoch || (newEpoch && epochs == numEpochs)) {
+        if (newEpoch || epochs == numEpochs) {
             log.info("Epoch " + epochs + ", number of batches completed " + lastBatch);
```

Comment:
Please remove this change because the condition is incorrect if we accept anytime epochs == numEpochs to say we have achieved an epoch. If anything it should just be newEpoch for the condition.